### PR TITLE
Issue #1575: Enable new MSI upgrade code to support proper ref counting for SxS install of HostFxr (1.1)

### PR DESCRIPTION
**Pull request by vivmishra**
- Enable HostFxrMsi to install side by side between releases and build-to-build. This fix is for 1.1.
- This change will now enable the HostFxrMsi to be generated with a new upgrade code with every build.

- Reference Issue: #1575

### CODE REVIEW

**Comment by vivmishra** [comment]
**Following scenario will work this change:**
1. Bad 1.0.4 -> Fixed 1.0.X -> Uninstall any --- No Issues. Ref counts correctly so install or uninstall in any order is fine. Same with 1.1.
2. Bad 1.0.4 -> Fixed 1.1.X -> Uninstall any --- No Issues. 1.0/1.1 HostFXRs now install correctly in different folders now with ref counting. Earlier 1.1 used to trample 1.0, and in reverse install order 1.0 hostfxr was never installed.
3. Bad 1.1.X -> Fixed 1.0.X->Uninstall any --- No Issues. 1.0/1.1 HostFXRs now install correctly in different folders now with ref counting.
4. Fixed 1.0.X -> Bad 1.0.4 -> Bad 1.1.X --- HostFXRs for 1.0 is busted as 1.1 tramples the same. Install order of bad 1.0.4 is a problem here.

**To summarize the fix:**
1. All the 4 scenarios are covered.
2. We have added a new variable LockedHostFxrMSIVersion (1.1.2) for the duration when the MSI package has to deviate from others. When we take a fix to the hostfxr.dll itself in future then we can revert to both MSI and other packages to contain the same version (1.1.3 or later) of the package for hostfxr.
3. The MSI build related variables have been renamed to correctly reflect the internal MSI version and the MSI file version.
4. For publishing we will continue to publish the packages to Azure blob to the same base version (1.1.0) folder.

**Comment by gkhanna79** [comment]
@dotnet-bot test OSX x64 Release Build please
